apps/src/studio/Item.js

```diff
@@ -358,3 +444,37 @@ Item.prototype.moveToNextPosition = function () {
   this.x = next.x;
   this.y = next.y;
 };
+
+/**
+ * Mark that we're colliding with object represented by key.
+ * Here, override base implemention to special on-collision logic for certain
+ * item classes.
+ * @param key A unique key representing the object we're colliding with
+ * @returns {boolean} True if collision is started, false if we're already colliding
+ * @override
+ */
+Item.prototype.startCollision = function (key) {
+  var newCollisionStarted = Item.superPrototype.startCollision.call(this, key);
+  if (newCollisionStarted) {
+    if (this.className === 'hazard') {
+      var actor = Studio.sprite[key];
+      if (actor) {
+        Studio.sprite[key].addAction(new spriteActions.FadeActor(2000));
```

Comment:
Agreed.  It should be pretty simple to push combinations of these actions onto an actor (and make them more configurable) although things like the shake action won't necessarily play nice with movement yet - needs more work to set up animation offsets or something.